04.ls/ls.rb

```diff
@@ -19,17 +19,22 @@
 PERMISSION_SEPARATOR = 512
 
 def main
-  options = { l: false }
+  options = { a: false, r: false, l: false }
   opt = OptionParser.new
+  opt.on('-a') { options[:l] = true }
+  opt.on('-r') { options[:l] = true }
   opt.on('-l') { options[:l] = true }
   argv = opt.parse(ARGV)
-
   path = argv[0] || './'
-  FileTest.directory?(path) or return
 
-  file_names = Dir.children(path).sort
-  filtered_file_names = file_names.reject { |name| name.start_with?('.') }
-  options[:l] ? display_file_names_with_long(path, filtered_file_names) : display_file_names(filtered_file_names)
+  file_names = fetch_file_names(path, options)
+  options[:l] ? display_file_names_with_long(path, file_names) : display_file_names(file_names)
+end
+
+def fetch_file_names(path, options)
+  file_names = Dir.glob('*', File::FNM_DOTMATCH, base: path).sort_by { |name| name.delete('.') }
+  filtered_file_names = options[:a] ? file_names : file_names.reject { |name| name.start_with?('.') }
+  options[:r] ? filtered_file_names.reverse : filtered_file_names
 end
 
 def display_file_names_with_long(path, file_names)
```

Comment:
withの後ろには名詞が来るべきなのでwith + 形容詞はちょっと変ですね。
そもそもwithではなくinを使って、

```suggestion
def display_file_names_in_long_format(path, file_names)
```

としてみるとか。
ただ、これだとちょっと長いので

```suggestion
def display_in_long_format(path, file_names)
```

とかでもいいかもしれません。
（file nameを表示することは引数のfile_namesで伝わる）